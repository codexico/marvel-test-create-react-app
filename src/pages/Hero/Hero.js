import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import { apiHeroByID } from '../../api/api';

import './Hero.css';

function Hero({ heroID }) {
  const [character, setCharacter] = useState();

  useEffect(() => {
    apiHeroByID(heroID).then(({ data }) => setCharacter(data.data.results[0]));
  }, [heroID]);

  return (
    <div className="Hero">
      {character && (
        <>
          <h1>{character.name}</h1>
          <p>{character.description}</p>
          <img
            src={`${character.thumbnail.path}/detail.${character.thumbnail.extension}`}
            alt={character.name}
          />

          <p>Quadrinhos {character.comics.available}</p>

          <p>Últimos lançamentos</p>
          {character.comics.items.map((comic, i) => {
            return (
              <div key={`comic-${heroID}-${i}`}>
                <p>{comic.name}</p>
              </div>
            );
          })}
        </>
      )}
    </div>
  );
}

Hero.propTypes = {
  heroID: PropTypes.number.isRequired,
};

export default Hero;
