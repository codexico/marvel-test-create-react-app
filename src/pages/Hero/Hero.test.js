import React from 'react';
import { render } from '@testing-library/react';

import Hero from './Hero';

test('Component renders correctly', () => {
  const { asFragment } = render(<Hero heroID={123} />);

  expect(asFragment()).toMatchSnapshot();
});
