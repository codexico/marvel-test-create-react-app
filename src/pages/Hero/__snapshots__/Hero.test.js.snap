// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Page Component renders correctly when hero has all data 1`] = `
<DocumentFragment>
  <div
    class="Hero"
  >
    <h1>
      Hulk
    </h1>
    <p>
      Caught in a gamma bomb explosion while trying to save the life of a teenager, Dr. Bruce Banner was transformed into the incredibly powerful creature called the Hulk. An all too often misunderstood hero, the angrier the Hulk gets, the stronger the Hulk gets.
    </p>
    <img
      alt="Hulk"
      src="http://i.annihil.us/u/prod/marvel/i/mg/5/a0/538615ca33ab0/detail.jpg"
    />
    <p>
      Quadrinhos 1666
    </p>
    <p>
      Últimos lançamentos
    </p>
    <div>
      <p>
        5 Ronin (Hardcover)
      </p>
    </div>
    <div>
      <p>
        5 Ronin (2010) #2
      </p>
    </div>
    <div>
      <p>
        5 Ronin (2010) #2 (BROOKS COVER)
      </p>
    </div>
    <div>
      <p>
        A+X (2012) #1
      </p>
    </div>
    <div>
      <p>
        A+X (2012) #7
      </p>
    </div>
    <div>
      <p>
        Absolute Carnage: Immortal Hulk (2019) #1
      </p>
    </div>
    <div>
      <p>
        Actor Presents Spider-Man and the Incredible Hulk (2003) #1
      </p>
    </div>
    <div>
      <p>
        Age of X: Universe (2011) #1
      </p>
    </div>
    <div>
      <p>
        Age of X: Universe (2011) #2
      </p>
    </div>
    <div>
      <p>
        All-New Savage She-Hulk (2009) #1
      </p>
    </div>
    <div>
      <p>
        All-New Savage She-Hulk (2009) #2
      </p>
    </div>
    <div>
      <p>
        Alpha Flight (1983) #29
      </p>
    </div>
    <div>
      <p>
        Alpha Flight (1983) #110
      </p>
    </div>
    <div>
      <p>
        Alpha Flight (1983) #111
      </p>
    </div>
    <div>
      <p>
        Alpha Flight (1983) #127
      </p>
    </div>
    <div>
      <p>
        The Amazing Spider-Man (1963) #14
      </p>
    </div>
    <div>
      <p>
        Amazing Spider-Man (1999) #667
      </p>
    </div>
    <div>
      <p>
        Amazing Spider-Man Annual (1964) #3
      </p>
    </div>
    <div>
      <p>
        Amazing Spider-Man Annual (1964) #12
      </p>
    </div>
    <div>
      <p>
        Amazing Spider-Man Annual (2011) #38
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hero Page Component renders correctly when hero has minimal data 1`] = `
<DocumentFragment>
  <div
    class="Hero"
  >
    <h1>
      Black Queen
    </h1>
    <p />
    <img
      alt="Black Queen"
      src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/detail.jpg"
    />
    <p>
      Quadrinhos 2
    </p>
    <p>
      Últimos lançamentos
    </p>
    <div>
      <p>
        Captain America (1968) #369
      </p>
    </div>
    <div>
      <p>
        X-Men Unlimited (1993) #33
      </p>
    </div>
  </div>
</DocumentFragment>
`;
